include/translate-option.php

```diff
@@ -191,17 +192,17 @@ protected function register_string_recursive( $context, $option, $values, $key )
 			$children = is_array( $key ) ? $key : array();
 
 			if ( count( $children ) ) {
+				$matcher = new PLL_Format_Util();
+
 				foreach ( $children as $name => $child ) {
 					if ( isset( $values[ $name ] ) ) {
 						$this->register_string_recursive( $context, $name, $values[ $name ], $child );
 						continue;
 					}
 
-					$pattern = '#^' . str_replace( '*', '(?:.+)', $name ) . '$#';
-
 					foreach ( $values as $n => $value ) {
 						// The first case could be handled by the next one, but we avoid calls to preg_match here.
```

Comment:
Removed in [1c322e5](https://github.com/polylang/polylang/pull/1404/commits/1c322e5a02d20933e2418cce1142ac6104f597ce).

Also, I added [7f7c672](https://github.com/polylang/polylang/pull/1404/commits/7f7c6725bf85d2157a308d8aee55321f7f260d82) to prevent doing `forech` if `$name` is not a format.